libtenzir/src/execution_node.cpp

```diff
@@ -904,9 +887,11 @@ auto exec_node(
     return exec_node_actor::behavior_type::make_empty_behavior();
   }
   self->state.weak_node = node;
-  self->attach_functor([name = self->state.op->name()] {
-    TENZIR_DEBUG("exec-node for {} shut down", name);
-  });
+  self->attach_functor(
+    [name = self->state.op->name(), metrics = self->state.metrics] {
+      TENZIR_DEBUG("exec-node for {} shut down", name);
+      metrics->emit();
```

Comment:
This makes the functor emit metrics for pipelines that may have been deleted already, e.g. creating an immediately starting pipeline with wrong syntax in the `pipeline-manager`.

To be fair, this also already happens for diagnostics sometimes, so this is more an FYI/TODO than an immediate FIXIT.

Reply:
I think that's okay. It's better than not having metrics in the case of an ungraceful shutdown.